src/overlays/actors/ovl_Demo_Geff/z_demo_geff.c

```diff
@@ -22,33 +46,180 @@ const ActorInit Demo_Geff_InitVars =
     (ActorFunc)DemoGeff_Update,
     (ActorFunc)DemoGeff_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/DemoGeff_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/DemoGeff_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_80977EA8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_80977F80.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_80978030.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_809780E0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_809781FC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_809782A0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_80978308.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_80978344.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_80978370.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_809783D4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/DemoGeff_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_809784D4.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/DemoGeff_Draw.s")
+extern UNK_TYPE D_06000EA0;
+
+static void DemoGeff_Destroy(DemoGeff* this, GlobalContext* globalCtx) {
+
+}
+
+static void DemoGeff_Init(DemoGeff* this, GlobalContext* globalCtx) {
+    if (this->actor.params < 0 || this->actor.params >= 9) {
+        osSyncPrintf(VT_FGCOL(RED) "Demo_Geff_Actor_ct:arg_dataがおかしい!!!!!!!!!!!!\n" VT_RST);
+        Actor_Kill(&this->actor);
+        return;
+    }
+    this->action = 0;
+    this->drawConfig = 0;
+}
+
+static void func_80977EA8(GlobalContext* globalCtx, u32 dlist) {
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    s16 pad;
+    Gfx* gfxArr[4];
+
+    func_800C6AC4(gfxArr, gfxCtx, "../z_demo_geff.c", 181);
+    func_80093D18(gfxCtx);
+
+    gSPMatrix(gfxCtx->polyOpa.p++, Matrix_NewMtx(gfxCtx, "../z_demo_geff.c", 183),
+              G_MTX_PUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(gfxCtx->polyOpa.p++, dlist);
+    gSPPopMatrix(gfxCtx->polyOpa.p++, G_MTX_MODELVIEW);
+
+    func_800C6B54(gfxArr, gfxCtx, "../z_demo_geff.c", 188);
+}
+
+static void func_80977F80(DemoGeff* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    s32 objBankIndex = this->objBankIndex;
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    s16 pad2;
+    Gfx* gfxArr[4];
+
+    func_800C6AC4(gfxArr, gfxCtx, "../z_demo_geff.c", 204);
+    
+    gSPSegment(gfxCtx->polyOpa.p++, 0x06, globalCtx->objectCtx.status[objBankIndex].segment);
+    gSegments[6] = PHYSICAL_TO_VIRTUAL(globalCtx->objectCtx.status[objBankIndex].segment);
+
+    if (!globalCtx) {
+
+    }
```

Comment:
I would keep this on a single line and annotate that it's "necessary to match", rather than actually serving a purpose.